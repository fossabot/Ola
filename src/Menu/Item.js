import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'
import { getElementType } from '../utils'

const Item = ({ as, className, variant, children, ...props }) => {
  const ElementType = getElementType(Item, { as: as, ...props })
  delete props['as']
  return (
    <li className={cx(variant && `ola_menu-${variant}`)}>
      <ElementType className={ cx('ola_menu-option', className) } {...props}>
        {children}
      </ElementType>
    </li>
  )
}


Item.defaultProps = {
  variant: null,
  as: 'span',
}

Item.propTypes = {
  /** Render Item with any html tag */
  as: PT.string,
  /** Item variants */
  variant: PT.oneOf(['separator']),
  /** Extra className */
  className: PT.string,
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired
}

export default Item
