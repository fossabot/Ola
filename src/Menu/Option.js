import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'
import { getElementType } from '../utils'

const Option = ({ as, className, separator, children, ...props }) => {
  const ElementType = getElementType(Option, { as: as, ...props })
  delete props['as']
  return (
    <li className={cx(separator && 'ola_menu-separator')}>
      <ElementType className={ cx('ola_menu-option', className) } {...props}>
        {children}
      </ElementType>
    </li>
  )
}


Option.defaultProps = {
  variant: null,
  as: 'a',
}

Option.propTypes = {
  /** Render Item with any html tag */
  as: PT.string,
  /** Separator */
  separator: PT.bool,
  /** Extra className */
  className: PT.string,
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired
}

export default Option
