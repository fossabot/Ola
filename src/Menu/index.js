import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'

const Menu = ({ className, children }) =>
  <ul className={cx('ola_menu', className)}>{children}</ul>

Menu.propTypes = {
  /** Extra className */
  className: PT.string,
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired
}

export default Menu
