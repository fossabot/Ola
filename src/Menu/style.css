.ola_menu {
    list-style: none;
    margin: 0;
    padding: 0;
}
.ola_menu-option {
    --color: var(--gray);
    --background: var(--white);

    font: var(--font-callout);
    color: var(--color);
    background: var(--background);
    display: flex;
    align-items: center;
    text-align: left;
    text-decoration: none;
    padding: var(--size-4) var(--size-6);
    outline: 0;
    border: none;
    box-sizing: border-box;
    width: 100%;
    cursor: pointer;

    &::-moz-focus-inner {
        border: 0;
    }

    &.is-nav {
        --color: var(--accent);
        border-top: solid 1px var(--gray-xlight);
        font-weight: var(--bold);
    }

    &.is-selected,
    &:hover,
    &:focus {
        --color: var(--black);
        --background: var(--gray-xlight);
    }

    &.is-selected {
        border-left: solid var(--size-1) var(--gray-light);
    }

    &:active {
        --color: var(--black);
        --background: var(--brand-focus);
    }

    & .ola_icon {
        flex: 0 0 auto;
        margin-right: var(--size-4);
    }
}
.ola_menu-separator {
    border-top: solid 1px var(--gray-light);
    margin-top: var(--size-3);
    padding-top: var(--size-3);
}
.ola_tooltip-content .ola_menu-option {
    margin-left: calc(var(--size-6) * -1);
    margin-right: calc(var(--size-6) * -1);
    width: calc(100% + var(--size-6) * 2);
}
