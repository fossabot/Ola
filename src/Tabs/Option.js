import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'
import { getElementType } from '../utils'

const TabOption = ({ as, className, children, selected, ...props }) => {
  const ElementType = getElementType(TabOption, { as: as, ...props })
  delete props['as']

  return (
    <li>
      <ElementType aria-current={selected} className={ cx(
        'ola_tab-option',
        className,
        {
          ['is-selected']: selected 
        }) } {...props}>
        {children}
      </ElementType>
    </li>
  )
}


TabOption.defaultProps = {
  as: 'button',
}

TabOption.propTypes = {
  /** Render Item with any html tag */
  as: PT.string,
  /** Selected */
  selected: PT.bool,
  /** Extra className */
  className: PT.string,
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired
}

export default TabOption
