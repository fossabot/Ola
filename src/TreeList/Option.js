import React from "react";
import { default as PT } from "prop-types";
import cx from "classnames";

const TreeListOption = ({ className, variant, children, ...props }) => {
  const styles = cx(
    "ola_treeList-option",
    { [`is-${variant}`]: variant },
    className
  );
  return (
    <li className={styles} {...props}>
      {children}
    </li>
  );
};

TreeListOption.defaultProps = {
  className: null,
  variant: null,
};

TreeListOption.propTypes = {
  /** Extra className */
  className: PT.string,
  /** TreeListOption variants */
  variant: PT.oneOf(["link"]),
  /** Childen nodes */
  children: PT.oneOfType([PT.string, PT.arrayOf(PT.node), PT.node]).isRequired,
};

export default TreeListOption;
