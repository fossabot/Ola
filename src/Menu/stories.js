import React from 'react'
import { storiesOf } from '@storybook/react'
import ButtonIcon from '../ButtonIcon'
import Icon from '../Icon'
import Tooltip from '../Tooltip'

import Menu from './'
import Option from './Option'

storiesOf('Menu')
  .add('Default', () => (
    <figure>
      <Menu>
        <Option>
          <Icon name="user" />
          <strong>Profile</strong>
        </Option>
        <Option href="#">
          <Icon name="help" />
          <strong>Help</strong>
        </Option>
        <Option separator>
          <Icon name="error" />
          <strong>Logout</strong>
        </Option>
      </Menu>
    </figure>
  ))
  .add('In a tooltip', () => (
    <figure>
      <Tooltip trigger={<ButtonIcon as="span"><Icon name="user" /></ButtonIcon>}>
        <Menu>
          <Option>
            <Icon name="user" />
            <strong>Profile</strong>
          </Option>
          <Option href="#">
            <Icon name="help" />
            <strong>Help</strong>
          </Option>
          <Option separator>
            <Icon name="error" />
            <strong>Logout</strong>
          </Option>
        </Menu>
      </Tooltip>
    </figure>
  ))
