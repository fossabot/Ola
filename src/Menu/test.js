import React from 'react'
import Menu from './'
import Option from './Option'
import renderer from 'react-test-renderer'

it('Default Menu with Options', () => {
  const tree = renderer
    .create(
      <Menu>
        <Option>Option 1</Option>
        <Option>Option 2</Option>
      </Menu>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Options are links', () => {
  const tree = renderer
    .create(
      <Menu>
        <Option href="link1">Option 1</Option>
        <Option href="link2">Option 2</Option>
      </Menu>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Default Menu with separator items', () => {
  const tree = renderer
    .create(
      <Menu>
        <Option>Option 1</Option>
        <Option separator>Option 2</Option>
        <Option>Option 3</Option>
        <Option separator>Option 4</Option>
      </Menu>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
