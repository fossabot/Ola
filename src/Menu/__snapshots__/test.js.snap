// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Menu with Options 1`] = `
<ul
  className="ola_menu"
>
  <li
    className=""
  >
    <a
      className="ola_menu-option"
      variant={null}
    >
      Option 1
    </a>
  </li>
  <li
    className=""
  >
    <a
      className="ola_menu-option"
      variant={null}
    >
      Option 2
    </a>
  </li>
</ul>
`;

exports[`Default Menu with separator items 1`] = `
<ul
  className="ola_menu"
>
  <li
    className=""
  >
    <a
      className="ola_menu-option"
      variant={null}
    >
      Option 1
    </a>
  </li>
  <li
    className="ola_menu-separator"
  >
    <a
      className="ola_menu-option"
      variant={null}
    >
      Option 2
    </a>
  </li>
  <li
    className=""
  >
    <a
      className="ola_menu-option"
      variant={null}
    >
      Option 3
    </a>
  </li>
  <li
    className="ola_menu-separator"
  >
    <a
      className="ola_menu-option"
      variant={null}
    >
      Option 4
    </a>
  </li>
</ul>
`;

exports[`Options are links 1`] = `
<ul
  className="ola_menu"
>
  <li
    className=""
  >
    <a
      className="ola_menu-option"
      href="link1"
      variant={null}
    >
      Option 1
    </a>
  </li>
  <li
    className=""
  >
    <a
      className="ola_menu-option"
      href="link2"
      variant={null}
    >
      Option 2
    </a>
  </li>
</ul>
`;
